application/src/main/java/org/togetherjava/tjbot/commands/reminder/RemindRoutine.java

```diff
@@ -0,0 +1,139 @@
+     * Creates a new instance.
+     *
+     * @param database the database that contains the pending reminders to send.
+     */
+    public RemindRoutine(@NotNull Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public @NotNull Schedule createSchedule() {
+        return new Schedule(ScheduleMode.FIXED_RATE, 0, SCHEDULE_INTERVAL_SECONDS,
+                TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void runRoutine(@NotNull JDA jda) {
+        Instant now = Instant.now();
+        database.write(context -> context.selectFrom(PENDING_REMINDERS)
+            .where(PENDING_REMINDERS.REMIND_AT.lessOrEqual(now))
+            .stream()
+            .forEach(pendingReminder -> {
+                sendReminder(jda, pendingReminder.getId(), pendingReminder.getGuildId(),
+                        pendingReminder.getChannelId(), pendingReminder.getAuthorId(),
+                        pendingReminder.getContent(), pendingReminder.getCreatedAt());
+                pendingReminder.delete();
+            }));
+    }
+
+    private static void sendReminder(@NotNull JDA jda, long id, long guildId, long channelId,
+            long authorId, @NotNull CharSequence content, @NotNull TemporalAccessor createdAt) {
+        RestAction<ReminderRoute> route = computeReminderRoute(jda, guildId, channelId, authorId);
+        sendReminderViaRoute(route, id, content, createdAt);
+    }
+
+    private static RestAction<ReminderRoute> computeReminderRoute(@NotNull JDA jda, long guildId,
+            long channelId, long authorId) {
+        // If guild and channel can still be found, send there
+        Guild guild = jda.getGuildById(guildId);
+        if (guild != null) {
+            TextChannel channel = guild.getTextChannelById(channelId);
+            if (channel != null) {
+                return createGuildReminderRoute(jda, authorId, channel);
+            }
+        }
+
+        // Otherwise, attempt to DM the user directly
+        return createDmReminderRoute(jda, authorId);
+    }
+
+    private static @NotNull RestAction<ReminderRoute> createGuildReminderRoute(@NotNull JDA jda,
+            long authorId, @NotNull TextChannel channel) {
+        return jda.retrieveUserById(authorId)
+            .onErrorMap(error -> null)
+            .map(author -> new ReminderRoute(channel, author,
+                    author == null ? null : author.getAsMention()));
+    }
+
+    private static @NotNull RestAction<ReminderRoute> createDmReminderRoute(@NotNull JDA jda,
+            long authorId) {
+        return jda.openPrivateChannelById(authorId)
+            .onErrorMap(error -> null)
+            .map(channel -> channel == null ? new ReminderRoute(null, null, null)
+                    : new ReminderRoute(channel, channel.getUser(),
+                            "(Sending your reminder directly, because I was unable to"
+                                    + " locate the original channel you wanted it to be send to)"));
```

Comment:
```suggestion
        return jda.openPrivateChannelById(authorId)
                .map(channel -> new ReminderRoute(channel, channel.getUser(),
                        "(Sending your reminder directly, because I was unable to"
                                + " locate the original channel you wanted it to be send to)"));
```